nncf/sparsity/magnitude/algo.py

```diff
@@ -17,11 +17,13 @@
 
 from nncf.algo_selector import COMPRESSION_ALGORITHMS
 from nncf.compression_method_api import CompressionAlgorithmController, CompressionLevel
+from nncf.initialization import DataLoaderBNAdaptationRunner
 from nncf.nncf_network import NNCFNetwork
 from nncf.sparsity.base_algo import BaseSparsityAlgoBuilder, BaseSparsityAlgoController, SparseModuleInfo
 from nncf.sparsity.layers import BinaryMask
 from nncf.sparsity.magnitude.functions import WEIGHT_IMPORTANCE_FUNCTIONS, calc_magnitude_binary_mask
 from nncf.sparsity.schedulers import SPARSITY_SCHEDULERS
+from nncf.structures import BNAdaptationInitArgs
 
 
 @COMPRESSION_ALGORITHMS.register('magnitude_sparsity')
```

Comment:
Is BN adaptation harmful for RB-sparsity? Why did you add it for magnitude one only?

Reply:
The sparsification of layers in RB sparsity is not manual like in the magnitude method. For the magnitude method, the adaptation is done once each time a binary mask is constructed. There's no specific point in the training pipeline for RB sparsity where it makes sense to do BN adaptation.